prometheus-to-sd/config/gce_config.go

```diff
@@ -84,10 +84,17 @@ func GetGceConfig(project, cluster, clusterLocation, zone, node string) (*GceCon
 		}
 	}
 
+	// instance/name endpoint is not available on the GKE metadata server.
+	// Try GCE instance/name endpoint. If error, try instance/hostname.
+	// If instance/hostname, remove domain to replicate instance/name.
 	if node == "" {
 		node, err = gce.InstanceName()
 		if err != nil {
-			return nil, fmt.Errorf("error while getting instance (node) name: %v", err)
+			node, err = gce.Hostname()
+			if err != nil {
+				return nil, fmt.Errorf("error while getting instance (node) name: %v", err)
+			}
+			node = strings.Split(node, ".")[0]
```

Comment:
Can you log a warning here? In case instance name isn't actually a part of the hostname and someone tries to debug this.

Reply:
@x13n done :)